### DISCUSSION TITLE
Q: BumpAssemblyVersions (BAV) which reference assemblies assume dependencies

### DISCUSSION THREAD

**Discussion by mwpowellhtx** [dotnet/msbuild#10192] [Q&A] [answered]
Hello,

I am evaluating my [`BumpAssemblyVersions`](https://github.com/mwpowellhtx/BumpAssemblyVersions) project, whether I need to update and pull forward appropriate forwards/backwards compatible [`MSBuild`](https://github.com/dotnet/msbuild) dependencies. I am assuming I do since the last known [`Microsoft.Build`](https://nuget.org/packages/Microsoft.Build) dependency at `1.8.0` was something like `15.9.X` (specifically `15.9.20`). Also `Engine`, `Framework`, `Utilities.Core`, couple of other odd ones as well. Yeah I know it has been a minute since I had to look at this.

Best guess, I need to rebuild with the updated versions. 

Question when I do that, is it backwards compatible with the prior versions? `15+`, `16+`, `17+`? Or do I need specifically built projects around those?

Thanks...

**Comment by AR-May** [comment, marked as answer]
Hello @mwpowellhtx! MSBuild and (non-experimental) MSBuild public API for the most part is backwards compatible. There were very few breaking changes, and they are announced with the release of MSBuild within Visual Studio or dotnet sdk. 

As for updating the version of MSBuild. It seems like you are authoring an MSBuild task. According to [this documentation](https://learn.microsoft.com/en-us/visualstudio/msbuild/tutorial-custom-task-code-generation?view=vs-2022#choose-the-msbuild-api-version-to-reference) the earlier version of MSBuild API you target, the older VS versions will support the task. And, because of backwards compatibility, newer versions will also be able to run it. So, unless your task does not work in the latest VS, I see no reason to upgrade.
